java/java.editor.base/src/org/netbeans/modules/java/editor/base/semantic/UnusedDetector.java

```diff
@@ -597,5 +619,49 @@ private void handleDeclaration(TreePath path) {
                 }
             }
         }
+
+        @Override
+        public Void visitLiteral(LiteralTree node, Void p) {
+            if (node.getKind() == Kind.STRING_LITERAL) {
+                allStringLiterals.add((String) ((LiteralTree) node).getValue());
```

Comment:
The whole PR is conclusive to me, this part I don't understand. My initial thought was, that you are trying to get the strings used in lookups, but these are already covered by the explicit invocation in `visitMethodInvocation`. Or is this to find "indirect" lookups? If so, should this customizable?

If I'm right - does this only match the unqualified name of a  method or a field?

Reply:
Sorry for late reply. Yes, this is for indirect lookups - if the lookup contains a method name, the `allStringLiterals` should not be used. This is all a heuristics (e.g. it never uses the `MethodType`, so if there are several private overloads, they will all be considered used, even those that will never be looked up). But, this is just a warning, I think it is fine to use sensible heuristics, and I am more fine with some false negatives (not found unused methods) than with false positives (used methods marked as unused).

If the code is in a recognized pattern (`$lookup.findSomething($className.class, ...)`, the class in which the lookup is performed will be taken into consideration. If the pattern is different, the type won't be considered, and more methods will be potentially marked as used. This is again mostly a heuristics that should find a relatively typical pattern.

This should all have zero effect on files which don't use `MethodHandles.Lookup`, which is a vast majority of files, and the proposed heuristics seemed like a reasonable tradeoff to me.

Reply:
Think you for the explanation. With that in mind it makes sense to me. I overlooked the combined effect of:

https://github.com/apache/netbeans/pull/5173/files/7c57e4b4d7faf2b094e67ffab43100182f4f45bb#diff-908c30e8d355bd77779d4bfffcd966667091f27f2ab26299889fcceb659478c6R643-R662

and

https://github.com/apache/netbeans/pull/5173/files/7c57e4b4d7faf2b094e67ffab43100182f4f45bb#diff-908c30e8d355bd77779d4bfffcd966667091f27f2ab26299889fcceb659478c6R297